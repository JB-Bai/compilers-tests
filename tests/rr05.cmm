int ksOWRB(struct Zpx4EtraRH o6XEWbU, float JRnai){
	struct qylOKX _7BOaBe;
	int HTwX = Iu5iRAB5() * MnXK_zaC / f3wyyN9VQ[060154] &&(219.945e-2) - 911.9 = tXM5vhEsV * 0010.80 / ! jl7Lub9 / q5zotal().eFd5vbwRZP;
	if(DFuJ8YdLQ(tr3o1 + wPKsFlGiT) - rghgW_z() - dYNBRS){
		struct{
		}
		yyLEu8, UT0CkJ3 = VNemEazHu(008.41E2 / PC8zclN7Oi || - 024, LENxh + - 1.427.wCkV2UFrlt, 0x2c) + ennT5HwGX;
		int RjSK13 = 42.822e+2 = cFvnTUb, Wpfb77h4NK;
		struct M7Q7ar{
		}
		HLWTHD[0530];
		BHYhMjq;
	}
	if(6.3178E-5){
		struct{
		}
		qq9MvkeTN3[0225][0x1Ef] = TFt41ln(), E_akXbONKn, rWcGAll[23224][0xDBd13], CoNVGeWT[74], sm6UGcKC7 = UVLz6Pr, kyF8StCbO;
		float Vm7FFWS;
		float Ruzbia[6928][3163], a2F1A = 2809;
		if(0037) 44.4332E8 - ! T4OzrzALL;
		if(0756) sCCQz3RKD((0711104 * ! 0xaDe || fXMxtx = ! 182.33e9 = 260 - Ojn9Y0() + 0622545), z9SPzc2iAh5);
		while(21.07) while(L5HUya6ru) XZY8zSr().Vngm56rppG6w || 03.008;
		return(0xAf5);
		return 191[0634];
	}
	{
		(529446).kdxEd6[- 00760];
		{
			return UAsh0j;
			khbUbxKh5;
		}
		0x6B1F / 8.64e-0;
		return 0xDeCd;
	}
	gFGpEFZe().hX9D05sLN / ttZ2Ah;
	if(937.450e1 && 667.19 - f_KNbYvCIP) while(- 01462 / 67.51e1) z3dy0it(077345);
	21.5637e-1;
}
float aXTGSvkYMS;
float RCHgI7Iw(struct gRrazy3 YXHRBBbq[0x28E], int f4ykFK_i, int LlmUdhWd[032574]){
	struct SrwyFZH9A{
	}
	H6Eml_Q, jHPa0K;
	SbVtxBMsNF[uEHhg];
}

