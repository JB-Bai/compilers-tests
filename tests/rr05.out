Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: ksOWRB
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              StructSpecifier (1)
                STRUCT
                Tag (1)
                  ID: Zpx4EtraRH
            VarDec (1)
              ID: o6XEWbU
          COMMA
          VarList (1)
            ParamDec (1)
              Specifier (1)
                TYPE: float
              VarDec (1)
                ID: JRnai
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              StructSpecifier (2)
                STRUCT
                Tag (2)
                  ID: qylOKX
            DecList (2)
              Dec (2)
                VarDec (2)
                  ID: _7BOaBe
            SEMI
          DefList (3)
            Def (3)
              Specifier (3)
                TYPE: int
              DecList (3)
                Dec (3)
                  VarDec (3)
                    ID: HTwX
                  ASSIGNOP
                  Exp (3)
                    Exp (3)
                      Exp (3)
                        Exp (3)
                          Exp (3)
                            ID: Iu5iRAB5
                            LP
                            RP
                          STAR
                          Exp (3)
                            ID: MnXK_zaC
                        DIV
                        Exp (3)
                          Exp (3)
                            ID: f3wyyN9VQ
                          LB
                          Exp (3)
                            INT: 24684
                          RB
                      AND
                      Exp (3)
                        Exp (3)
                          LP
                          Exp (3)
                            FLOAT: 2.199450
                          RP
                        MINUS
                        Exp (3)
                          FLOAT: 911.900024
                    ASSIGNOP
                    Exp (3)
                      Exp (3)
                        Exp (3)
                          Exp (3)
                            ID: tXM5vhEsV
                          STAR
                          Exp (3)
                            FLOAT: 10.800000
                        DIV
                        Exp (3)
                          NOT
                          Exp (3)
                            ID: jl7Lub9
                      DIV
                      Exp (3)
                        Exp (3)
                          ID: q5zotal
                          LP
                          RP
                        DOT
                        ID: eFd5vbwRZP
              SEMI
        StmtList (4)
          Stmt (4)
            IF
            LP
            Exp (4)
              Exp (4)
                Exp (4)
                  ID: DFuJ8YdLQ
                  LP
                  Args (4)
                    Exp (4)
                      Exp (4)
                        ID: tr3o1
                      PLUS
                      Exp (4)
                        ID: wPKsFlGiT
                  RP
                MINUS
                Exp (4)
                  ID: rghgW_z
                  LP
                  RP
              MINUS
              Exp (4)
                ID: dYNBRS
            RP
            Stmt (4)
              CompSt (4)
                LC
                DefList (5)
                  Def (5)
                    Specifier (5)
                      StructSpecifier (5)
                        STRUCT
                        LC
                        RC
                    DecList (7)
                      Dec (7)
                        VarDec (7)
                          ID: yyLEu8
                      COMMA
                      DecList (7)
                        Dec (7)
                          VarDec (7)
                            ID: UT0CkJ3
                          ASSIGNOP
                          Exp (7)
                            Exp (7)
                              ID: VNemEazHu
                              LP
                              Args (7)
                                Exp (7)
                                  Exp (7)
                                    Exp (7)
                                      FLOAT: 841.000000
                                    DIV
                                    Exp (7)
                                      ID: PC8zclN7Oi
                                  OR
                                  Exp (7)
                                    MINUS
                                    Exp (7)
                                      INT: 20
                                COMMA
                                Args (7)
                                  Exp (7)
                                    Exp (7)
                                      ID: LENxh
                                    PLUS
                                    Exp (7)
                                      MINUS
                                      Exp (7)
                                        Exp (7)
                                          FLOAT: 1.427000
                                        DOT
                                        ID: wCkV2UFrlt
                                  COMMA
                                  Args (7)
                                    Exp (7)
                                      INT: 44
                              RP
                            PLUS
                            Exp (7)
                              ID: ennT5HwGX
                    SEMI
                  DefList (8)
                    Def (8)
                      Specifier (8)
                        TYPE: int
                      DecList (8)
                        Dec (8)
                          VarDec (8)
                            ID: RjSK13
                          ASSIGNOP
                          Exp (8)
                            Exp (8)
                              FLOAT: 4282.200195
                            ASSIGNOP
                            Exp (8)
                              ID: cFvnTUb
                        COMMA
                        DecList (8)
                          Dec (8)
                            VarDec (8)
                              ID: Wpfb77h4NK
                      SEMI
                    DefList (9)
                      Def (9)
                        Specifier (9)
                          StructSpecifier (9)
                            STRUCT
                            OptTag (9)
                              ID: M7Q7ar
                            LC
                            RC
                        DecList (11)
                          Dec (11)
                            VarDec (11)
                              VarDec (11)
                                ID: HLWTHD
                              LB
                              INT: 344
                              RB
                        SEMI
                StmtList (12)
                  Stmt (12)
                    Exp (12)
                      ID: BHYhMjq
                    SEMI
                RC
          StmtList (14)
            Stmt (14)
              IF
              LP
              Exp (14)
                FLOAT: 0.000063
              RP
              Stmt (14)
                CompSt (14)
                  LC
                  DefList (15)
                    Def (15)
                      Specifier (15)
                        StructSpecifier (15)
                          STRUCT
                          LC
                          RC
                      DecList (17)
                        Dec (17)
                          VarDec (17)
                            VarDec (17)
                              VarDec (17)
                                ID: qq9MvkeTN3
                              LB
                              INT: 149
                              RB
                            LB
                            INT: 495
                            RB
                          ASSIGNOP
                          Exp (17)
                            ID: TFt41ln
                            LP
                            RP
                        COMMA
                        DecList (17)
                          Dec (17)
                            VarDec (17)
                              ID: E_akXbONKn
                          COMMA
                          DecList (17)
                            Dec (17)
                              VarDec (17)
                                VarDec (17)
                                  VarDec (17)
                                    ID: rWcGAll
                                  LB
                                  INT: 23224
                                  RB
                                LB
                                INT: 900371
                                RB
                            COMMA
                            DecList (17)
                              Dec (17)
                                VarDec (17)
                                  VarDec (17)
                                    ID: CoNVGeWT
                                  LB
                                  INT: 74
                                  RB
                              COMMA
                              DecList (17)
                                Dec (17)
                                  VarDec (17)
                                    ID: sm6UGcKC7
                                  ASSIGNOP
                                  Exp (17)
                                    ID: UVLz6Pr
                                COMMA
                                DecList (17)
                                  Dec (17)
                                    VarDec (17)
                                      ID: kyF8StCbO
                      SEMI
                    DefList (18)
                      Def (18)
                        Specifier (18)
                          TYPE: float
                        DecList (18)
                          Dec (18)
                            VarDec (18)
                              ID: Vm7FFWS
                        SEMI
                      DefList (19)
                        Def (19)
                          Specifier (19)
                            TYPE: float
                          DecList (19)
                            Dec (19)
                              VarDec (19)
                                VarDec (19)
                                  VarDec (19)
                                    ID: Ruzbia
                                  LB
                                  INT: 6928
                                  RB
                                LB
                                INT: 3163
                                RB
                            COMMA
                            DecList (19)
                              Dec (19)
                                VarDec (19)
                                  ID: a2F1A
                                ASSIGNOP
                                Exp (19)
                                  INT: 2809
                          SEMI
                  StmtList (20)
                    Stmt (20)
                      IF
                      LP
                      Exp (20)
                        INT: 31
                      RP
                      Stmt (20)
                        Exp (20)
                          Exp (20)
                            FLOAT: 4443319808.000000
                          MINUS
                          Exp (20)
                            NOT
                            Exp (20)
                              ID: T4OzrzALL
                        SEMI
                    StmtList (21)
                      Stmt (21)
                        IF
                        LP
                        Exp (21)
                          INT: 494
                        RP
                        Stmt (21)
                          Exp (21)
                            ID: sCCQz3RKD
                            LP
                            Args (21)
                              Exp (21)
                                LP
                                Exp (21)
                                  Exp (21)
                                    Exp (21)
                                      Exp (21)
                                        INT: 234052
                                      STAR
                                      Exp (21)
                                        NOT
                                        Exp (21)
                                          INT: 2782
                                    OR
                                    Exp (21)
                                      ID: fXMxtx
                                  ASSIGNOP
                                  Exp (21)
                                    Exp (21)
                                      NOT
                                      Exp (21)
                                        FLOAT: 182329999360.000000
                                    ASSIGNOP
                                    Exp (21)
                                      Exp (21)
                                        Exp (21)
                                          INT: 260
                                        MINUS
                                        Exp (21)
                                          ID: Ojn9Y0
                                          LP
                                          RP
                                      PLUS
                                      Exp (21)
                                        INT: 206181
                                RP
                              COMMA
                              Args (21)
                                Exp (21)
                                  ID: z9SPzc2iAh5
                            RP
                          SEMI
                      StmtList (22)
                        Stmt (22)
                          WHILE
                          LP
                          Exp (22)
                            FLOAT: 21.070000
                          RP
                          Stmt (22)
                            WHILE
                            LP
                            Exp (22)
                              ID: L5HUya6ru
                            RP
                            Stmt (22)
                              Exp (22)
                                Exp (22)
                                  Exp (22)
                                    ID: XZY8zSr
                                    LP
                                    RP
                                  DOT
                                  ID: Vngm56rppG6w
                                OR
                                Exp (22)
                                  FLOAT: 3.008000
                              SEMI
                        StmtList (23)
                          Stmt (23)
                            RETURN
                            Exp (23)
                              LP
                              Exp (23)
                                INT: 2805
                              RP
                            SEMI
                          StmtList (24)
                            Stmt (24)
                              RETURN
                              Exp (24)
                                Exp (24)
                                  INT: 191
                                LB
                                Exp (24)
                                  INT: 412
                                RB
                              SEMI
                  RC
            StmtList (26)
              Stmt (26)
                CompSt (26)
                  LC
                  StmtList (27)
                    Stmt (27)
                      Exp (27)
                        Exp (27)
                          Exp (27)
                            LP
                            Exp (27)
                              INT: 529446
                            RP
                          DOT
                          ID: kdxEd6
                        LB
                        Exp (27)
                          MINUS
                          Exp (27)
                            INT: 496
                        RB
                      SEMI
                    StmtList (28)
                      Stmt (28)
                        CompSt (28)
                          LC
                          StmtList (29)
                            Stmt (29)
                              RETURN
                              Exp (29)
                                ID: UAsh0j
                              SEMI
                            StmtList (30)
                              Stmt (30)
                                Exp (30)
                                  ID: khbUbxKh5
                                SEMI
                          RC
                      StmtList (32)
                        Stmt (32)
                          Exp (32)
                            Exp (32)
                              INT: 27423
                            DIV
                            Exp (32)
                              FLOAT: 8.640000
                          SEMI
                        StmtList (33)
                          Stmt (33)
                            RETURN
                            Exp (33)
                              INT: 57037
                            SEMI
                  RC
              StmtList (35)
                Stmt (35)
                  Exp (35)
                    Exp (35)
                      Exp (35)
                        ID: gFGpEFZe
                        LP
                        RP
                      DOT
                      ID: hX9D05sLN
                    DIV
                    Exp (35)
                      ID: ttZ2Ah
                  SEMI
                StmtList (36)
                  Stmt (36)
                    IF
                    LP
                    Exp (36)
                      Exp (36)
                        FLOAT: 9374.500000
                      AND
                      Exp (36)
                        Exp (36)
                          FLOAT: 667.190002
                        MINUS
                        Exp (36)
                          ID: f_KNbYvCIP
                    RP
                    Stmt (36)
                      WHILE
                      LP
                      Exp (36)
                        Exp (36)
                          MINUS
                          Exp (36)
                            INT: 818
                        DIV
                        Exp (36)
                          FLOAT: 675.099976
                      RP
                      Stmt (36)
                        Exp (36)
                          ID: z3dy0it
                          LP
                          Args (36)
                            Exp (36)
                              INT: 32485
                          RP
                        SEMI
                  StmtList (37)
                    Stmt (37)
                      Exp (37)
                        FLOAT: 2.156370
                      SEMI
        RC
    ExtDefList (39)
      ExtDef (39)
        Specifier (39)
          TYPE: float
        ExtDecList (39)
          VarDec (39)
            ID: aXTGSvkYMS
        SEMI
      ExtDefList (40)
        ExtDef (40)
          Specifier (40)
            TYPE: float
          FunDec (40)
            ID: RCHgI7Iw
            LP
            VarList (40)
              ParamDec (40)
                Specifier (40)
                  StructSpecifier (40)
                    STRUCT
                    Tag (40)
                      ID: gRrazy3
                VarDec (40)
                  VarDec (40)
                    ID: YXHRBBbq
                  LB
                  INT: 654
                  RB
              COMMA
              VarList (40)
                ParamDec (40)
                  Specifier (40)
                    TYPE: int
                  VarDec (40)
                    ID: f4ykFK_i
                COMMA
                VarList (40)
                  ParamDec (40)
                    Specifier (40)
                      TYPE: int
                    VarDec (40)
                      VarDec (40)
                        ID: LlmUdhWd
                      LB
                      INT: 13692
                      RB
            RP
          CompSt (40)
            LC
            DefList (41)
              Def (41)
                Specifier (41)
                  StructSpecifier (41)
                    STRUCT
                    OptTag (41)
                      ID: SrwyFZH9A
                    LC
                    RC
                DecList (43)
                  Dec (43)
                    VarDec (43)
                      ID: H6Eml_Q
                  COMMA
                  DecList (43)
                    Dec (43)
                      VarDec (43)
                        ID: jHPa0K
                SEMI
            StmtList (44)
              Stmt (44)
                Exp (44)
                  Exp (44)
                    ID: SbVtxBMsNF
                  LB
                  Exp (44)
                    ID: uEHhg
                  RB
                SEMI
            RC
